combat-trainer.lic

```diff
@@ -4054,18 +4069,118 @@ class GameState
+  #
+  # @param skill
+  #   The weapon skill to train. We infer the weapon to wield based on this.
+  # @param hand
+  #   Which hand to equip the weapon into?
+  #   To equip offhand, specify 'left' or 'offhand'.
+  # @returns name of the wielded weapon
+  def wield_weapon(skill, hand = 'main')
     return if skill.eql?('Brawling')
     return if skill.eql?('Tactics')
 
-    @equipment_manager.wield_weapon_offhand(weapon_training[skill], skill)
-    weapon_training[skill]
+    hand = 'offhand' if skill =~ /Offhand Weapon/i
+    weapon_name = weapon_training[skill]
+    echo "wield_weapon: weapon_name=#{weapon_name}, skill=#{skill}, hand=#{hand}" if $debug_mode_ct
+    if is_damaris_weapon?(weapon_name)
+      return wield_damaris_weapon(weapon_name, hand)
+    else
+      if hand =~ /^(left|off)/i
+        @equipment_manager.wield_weapon_offhand?(weapon_name, skill)
+      else
+        @equipment_manager.wield_weapon?(weapon_name, skill)
+      end
+    end
+    return weapon_name
+  end
+
+  def wield_offhand_weapon(skill)
+    return wield_weapon(skill, 'offhand')
+  end
+
+  # Assumes the weapon name is a Damaris weapon.
+  # Designed to be called from `wield_weapon(skill)`.
+  #
+  # @param weapon_name
+  #   The <adj> + <noun> of the weapon to wield as configured in your gear.
+  # @param hand
+  #   Which hand to equip the weapon into?
+  #   To equip offhand, specify 'left' or 'offhand'.
+  # @returns name of the wielded weapon
+  def wield_damaris_weapon(weapon_name, hand = 'main')
+    # We need to wield based on the weapon's last known state then turn it to the desired noun.
+    damaris_weapon_group = get_damaris_weapon_group(weapon_name)
+    damaris_weapon_state = get_damaris_weapon_state(damaris_weapon_group)
+    echo "wield_damaris_weapon: weapon_to_wield=#{weapon_name}, damaris_group=#{damaris_weapon_group}, damaris_state=#{damaris_weapon_state}, hand=#{hand}" if $debug_mode_ct
+    if hand =~ /^(left|off)/i
+      @equipment_manager.wield_weapon_offhand?(damaris_weapon_state)
+    else
+      @equipment_manager.wield_weapon?(damaris_weapon_state)
+    end
+    # Damaris weapons can be turned until a certain date then need an infuser stone for more time.
+    # Only if we successfully turn the weapon to the new noun do we update the damaris weapon state for the group.
+    # Otherwise combat-trainer will be confused thinking we're using a different weapon than we really are.
+    if @equipment_manager.turn_to_weapon?(DRC.get_noun(damaris_weapon_state), DRC.get_noun(weapon_name))
+      set_damaris_weapon_state(damaris_weapon_group, weapon_name)
+      return weapon_name # turn successful, return the new weapon name
+    else
+      DRC.message("Unable to turn #{damaris_weapon_state} into #{weapon_name}. Is the transition possible? Does it require an infuser stone? STUDY the weapon for more details.")
+      return damaris_weapon_state # unable to turn, return unchanged weapon name
```

Comment:
Should we remove this weapon skill from the list of trainables skills since we can't get a weapon to train it?

We do something like this in some other places.

Reply:
I can see the merits of doing that, but I don't have the energy or motivation to tackle it in this PR at the moment.

If the code gets to the "Unable to turn {weapon}..." error, then I think either:
1. Their weapon's turn time has expired and they need to apply another infuser stone
2. Their Damaris weapons YAML config is incorrect

For regular day-to-day hunting, this block should not be happening.

I'm inclined to wait til people say they hit this and it's a problem for them. And since I've made several follow up commits per code review, we'll of course reach out again for more testers.

Thanks